internal/controller/helmchart_controller.go

```diff
@@ -510,6 +510,8 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		tlsConfig     *tls.Config
 		authenticator authn.Authenticator
 		keychain      authn.Keychain
+		tlsLoginOpt   helmreg.LoginOption
```

Comment:
unfortunately both are needed. We can have a tls config when constructing the client because we can use the client without login (internally it has 2 clients, and http client for crud ops and an authorizer for login). When we want to login over https, the authorizer requires a tlsConfig to be passed and the only way currently is by passing cert files path.